src/index.js

```diff
@@ -78,7 +79,35 @@ class Artifact {
       url = this.__queue.buildUrl(this.__queue.getArtifact, taskId, runId, name);
     }
 
-    await this.__client.downloadUrl({url, output: filename});
+    let {body, headers, statusCode, statusMessage} = await this.__client.runner.run({
+      req: {
+        url,
+        method: 'GET',
+        headers: {},
+      },
+      body: '',
+    });
+
+    // We only want to follow specific redirects
+    if (![301, 302, 303, 307, 308].includes(statusCode)) {
+      throw new Error('Redirect was not a valid status code (' + statusCode + ')');
+    }
+
+    let artifactUrl = headers.location;
+
+    if (typeof artifactUrl !== 'string') {
+      throw new Error('Redirect has missing Location key');
+    }
+
+    // We only want to parse the URL to ensure that it's a valid URL.  We're
+    // going to avoid parsing and reserializing because that could lead to
+    // subtle errors which we really don't have any reason to want to introduce
+    // here
+    if (urllib.parse(artifactUrl).protocol !== 'https:') {
```

Comment:
Not an issue to fix, but I'm surprised that the ":" is part of the protocol.

Reply:
yah... me too!

```javascript
> url.parse('https://www.google.com').protocol
'https:'
```